odoo/http.py

```diff
@@ -308,7 +308,10 @@ def redirect(self, location, code=303, local=True):
         if local:
             location = urls.url_parse(location).replace(scheme='', netloc='').to_url()
         if request and request.db:
-            return request.registry['ir.http']._redirect(location, code)
+            try:
+                return request.registry['ir.http']._redirect(location, code)
+            except psycopg2.OperationalError:  # this is a quick fix waiting for https://github.com/odoo/odoo/pull/54102
+                pass
```

Comment:
Create a database, connect to it, drop in the database manager: an error appears saying that the db does not exist.
This fix is a quick solution to have green tests, but not checking the drop response could be another solution.

Reply:
I'd like to suggest this patch to avoid the error, instead of the try/except in http.py:

```patch
diff --git addons/web/controllers/main.py addons/web/controllers/main.py
index fb81913c5a33..40de7beea21b 100644
--- addons/web/controllers/main.py
+++ addons/web/controllers/main.py
@@ -1123,6 +1123,7 @@ class Database(http.Controller):
         try:
             dispatch_rpc('db','drop', [master_pwd, name])
             request._cr = None  # dropping a database leads to an unusable cursor
+            request.disable_db = True
             return request.redirect('/web/database/manager')
         except Exception as e:
             error = "Database deletion error: %s" % (str(e) or repr(e))
```

An alternative to setting `disable_db` is to discard `request.session.db`, either works for me, same purpose. The idea is the same as for the discard of the cursor: avoid reusing a connection to a database that has disappeared.